test/e2e/e2e_test.go

```diff
@@ -240,6 +248,231 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
 				}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(wantPendingWorkloads, cmpopts.IgnoreFields(metav1.ObjectMeta{}, "Name")))
 			})
 		})
+
+		ginkgo.It("Should allow fetching information about pending workloads in LocalQueue", func() {
+			ginkgo.By("Verify there are zero pending workloads", func() {
+				gomega.Eventually(func() int {
+					info, err := visibilityClient.LocalQueues(nsA.Name).GetPendingWorkloadsSummary(ctx, localQueueA.Name, metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(info.Items)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(0))
+			})
+
+			ginkgo.By("Schedule a job which is pending due to lower priority", func() {
+				sampleJob2 = testingjob.MakeJob("test-job-2", nsA.Name).
+					Queue(localQueueA.Name).
+					Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"1ms"}).
+					Request(corev1.ResourceCPU, "1").
+					PriorityClass(lowPriorityClass.Name).
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, sampleJob2)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Verify there is one pending workload", func() {
+				gomega.Eventually(func() int {
+					info, err := visibilityClient.LocalQueues(nsA.Name).GetPendingWorkloadsSummary(ctx, localQueueA.Name, metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(info.Items)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(1))
+			})
+
+			ginkgo.By("Await for pods to be running", func() {
+				gomega.Eventually(func() int {
+					createdJob := &batchv1.Job{}
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(blockingJob), createdJob)).Should(gomega.Succeed())
+					return int(*createdJob.Status.Ready)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(1))
+			})
+
+			ginkgo.By("Terminate execution of the first workload to release the quota", func() {
+				gomega.Expect(util.DeleteAllPodsInNamespace(ctx, k8sClient, nsA)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Verify there are zero pending workloads, after the second workload is admitted", func() {
+				gomega.Eventually(func() int {
+					info, err := visibilityClient.LocalQueues(nsA.Name).GetPendingWorkloadsSummary(ctx, localQueueA.Name, metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(info.Items)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(0))
+			})
+		})
+
+		ginkgo.It("Should allow fetching information about position of pending workloads from different LocalQueues", func() {
```

Comment:
Can we move this and "Should allow fetching information about position of pending workloads from different LocalQueues from different Namespaces" to integration tests?

I'd like to make e2e lightweight.

Reply:
Currently, integration tests for the visibility server are under development. I definitely agree with the idea of keeping e2e as lightweight as possible, but I suggest keeping them as they are until we have integration tests ready.

Reply:
Yea,  @PBundyra already had one attempt to add integration tests, but it proved to be challenging, because it requires setting up kubeconfig / certificates for communication between the main API server and the visibility extension. Also, there is no good example, because the metrics-server only uses e2e tests. 

So, we went with e2e tests until the challenges with integration tests are figured out.

Also, me put an effort into running the tests as quickly as possible, in particular, we don't use the 5s `sleep` if unnecessary.

Reply:
Thanks for the clarification. Introducing integration tests seems to be hard work. So we can work on these follow-ups.